codex-rs/core/src/path_utils.rs

```diff
@@ -0,0 +1,159 @@
+use std::path::Path;
+use std::path::PathBuf;
+
+use crate::env;
+
+pub fn normalize_for_path_comparison(path: &Path) -> std::io::Result<PathBuf> {
+    let canonical = path.canonicalize()?;
+    Ok(normalize_for_wsl(canonical))
+}
+
+fn normalize_for_wsl(path: PathBuf) -> PathBuf {
+    if !env::is_wsl() {
```

Comment:
Can you refactor this so that what happens when `env::is_wsl()` is true is its own function that you exercise in unit tests so you don't need to `std::env::set_var()` and we can remove the `unsafe` blocks?